src/icq/system.rs

```diff
@@ -53,27 +53,65 @@ impl ICQSystem {
             match purple_message {
                 PurpleMessage::Login(account_info) => self.login(account_info).await,
             }
-            self.ping().await
         }
     }
 
-    async fn ping(&mut self) {
-        self.tx.send(SystemMessage::Ping).await;
-    }
-
     async fn login(&mut self, account_info: AccountInfo) {
         log::debug!("login");
-        match protocol::register(&account_info.phone_number, || {
-            log::debug!("read_code");
-            self.read_code(&account_info.account)
-        })
-        .await
-        {
-            Ok(_) => (),
-            Err(error) => {
-                log::error!("Failed to register account: {:?}", error);
+        let mut registered_account_info = {
+            account_info
+                .account
+                .proxy(&mut self.tx)
+                .exec(|account| {
+                    let token = account.get_string("token", "");
+                    if token == "" {
+                        None
+                    } else {
+                        Some(protocol::RegisteredAccountInfo {
```

Comment:
Should this object be called RegisteredAccountInfo? The name makes me think this is the object returned on account registration and that it would contain the uin and things like that but it looks like this contains session data instead.

Reply:
Well, this contains all the data we got from registering and needed to connect afterward. Any other suggestion?

Reply:
Is this the date you get from registering or the data you get from logging in?

Aren't there two different payloads?

Reply:
Is there a difference between the data returned by *login* and *register*?

If this is the same data as login, just name it "ConnectionInfo" or "SessionData", or something similar.

If there is a difference between the two, you can have a "RegistrationData" and a "ConnectionInfo"

Reply:
This data is created only once and should persist. LoginData (and UIN) is fetched through `startSession`. I can use `RegistrationData` and `SessionData`